nautobot/circuits/views.py

```diff
@@ -3,277 +3,282 @@
     queryset = Circuit.objects.all()
-    model_form = forms.CircuitForm
-    template_name = "circuits/circuit_edit.html"
-
-
-class CircuitDeleteView(generic.ObjectDeleteView):
-    queryset = Circuit.objects.all()
-
-
-class CircuitBulkImportView(generic.BulkImportView):
-    queryset = Circuit.objects.all()
-    model_form = forms.CircuitCSVForm
     table = tables.CircuitTable
+    form = forms.CircuitForm
+    filterset = filters.CircuitFilterSet
+    filterset_form = forms.CircuitFilterForm
+    import_form = forms.CircuitCSVForm
+    bulk_edit_form = forms.CircuitBulkEditForm
+    lookup_field = "pk"
+
+    def get_extra_context(self, request, view_type, instance):
+        if view_type == "detail":
+            # A-side termination
+            termination_a = (
+                CircuitTermination.objects.restrict(request.user, "view")
+                .prefetch_related("site__region")
+                .filter(circuit=instance, term_side=CircuitTerminationSideChoices.SIDE_A)
+                .first()
+            )
+            if (
+                termination_a
+                and termination_a.connected_endpoint
+                and hasattr(termination_a.connected_endpoint, "ip_addresses")
+            ):
+                termination_a.ip_addresses = termination_a.connected_endpoint.ip_addresses.restrict(
+                    request.user, "view"
+                )
+
+            # Z-side termination
+            termination_z = (
+                CircuitTermination.objects.restrict(request.user, "view")
+                .prefetch_related("site__region")
+                .filter(circuit=instance, term_side=CircuitTerminationSideChoices.SIDE_Z)
+                .first()
+            )
+            if (
+                termination_z
+                and termination_z.connected_endpoint
+                and hasattr(termination_z.connected_endpoint, "ip_addresses")
+            ):
+                termination_z.ip_addresses = termination_z.connected_endpoint.ip_addresses.restrict(
+                    request.user, "view"
+                )
+
+            return {
+                "termination_a": termination_a,
+                "termination_z": termination_z,
+            }
+        elif view_type == "list":
+            return {}
+        elif view_type == "bulk_edit":
+            return {}
+        else:
+            return {}
+
+
+class CircuitViewSetRouter(NautobotRouter):
+    viewset = CircuitViewSet
+
+    def __init__(self, prefix, basename):
+        super().__init__(prefix, basename)
+
+
+class CircuitTerminationViewset(NautobotViewSet):
+    model = CircuitTermination
+    queryset = CircuitTermination.objects.all()
+    form = forms.CircuitTerminationForm
+    lookup_field = "pk"
 
+    def alter_obj_for_edit(self, obj, request, url_args, url_kwargs):
+        if "circuit" in url_kwargs:
+            obj.circuit = get_object_or_404(Circuit, pk=url_kwargs["circuit"])
+        return obj
```

Comment:
Why `alter_obj_for_edit` and not just `alter_obj`? When would you be altering an object if not for edit?